manifests/server/validate_db.pp

```diff
@@ -2,12 +2,12 @@
 #
 # @api private
 class puppetdb::server::validate_db (
-  $database_host       = $puppetdb::params::database_host,
-  $database_port       = $puppetdb::params::database_port,
-  $database_username   = $puppetdb::params::database_username,
-  Variant[String[1], Sensitive[String[1]]] $database_password = $puppetdb::params::database_password,
-  $database_name       = $puppetdb::params::database_name,
-  $jdbc_ssl_properties = $puppetdb::params::jdbc_ssl_properties,
+  Stdlib::Host                                       $database_host       = $puppetdb::params::database_host,
+  Variant[Stdlib::Port::User, Pattern[/\A[0-9]+\Z/]] $database_port       = $puppetdb::params::database_port,
+  String[1]                                          $database_username   = $puppetdb::params::database_username,
+  Variant[String[1], Sensitive[String[1]]]           $database_password   = $puppetdb::params::database_password,
+  String[1]                                          $database_name       = $puppetdb::params::database_name,
+  Variant[String, Boolean]                           $jdbc_ssl_properties = $puppetdb::params::jdbc_ssl_properties,
```

Comment:
This is interesting.  The default value in `params.pp` seems to be `''`, but we also compare it to `false` so maybe this can be something like:

```suggestion
  Variant[String[0], Boolean[false]]                 $jdbc_ssl_properties = $puppetdb::params::jdbc_ssl_properties,
```

Which should probably be simplified to something like this in another PR because it breaks backwards compatibility:
```suggestion
  Optional[String[1]]                                $jdbc_ssl_properties = $puppetdb::params::jdbc_ssl_properties,
```

Reply:
As recommended I've set the lower bounds to `String[0]` to allow an empty value `''` and the `Boolean` type to allow only `false`. I agree, it would be worth revisiting in another PR to refactor the param to optional and perhaps to use `Pattern` to validate the value starts with `?` as an expected value would be `?ssl=true`.